src/components/IconWithTooltip/IconWithTooltip.test.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { render, screen, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+import IconWithTooltip from './index';
+import useWindowSize from '../../hooks/useWindowSize';
+
+jest.mock('./../../hooks/useWindowSize');
+
+
+const defaultAltText = 'infoooo';
+const defaultTooltipText = 'Tooool';
+const DEFAULT_PROPS = {
+  icon: faInfoCircle,
+  altText: defaultAltText,
+  tooltipText: defaultTooltipText,
+};
+
+describe('<IconWithTooltip />', () => {
+  it('renders the icon passed to it with alt text', () => {
+    useWindowSize.mockReturnValue({ width: 800 });
+    const { container } = render(<IconWithTooltip {...DEFAULT_PROPS} />);
+    expect(container.querySelector(`[data-icon=${faInfoCircle.iconName}]`)).toBeTruthy();
+  });
+  [
+    { windowSize: 800, expectedLocation: 'right' },
+    { windowSize: 700, expectedLocation: 'top' },
+  ].forEach((data) => {
+    it(`renders the tooltip on the ${data.expectedLocation} for ${data.windowSize}px screen`, () => {
+      useWindowSize.mockReturnValue({ width: data.windowSize });
+      const { container } = render(<IconWithTooltip {...DEFAULT_PROPS} />);
+      const icon = container.querySelector(`[data-icon=${faInfoCircle.iconName}]`);
+      expect(icon).toBeTruthy();
```

Comment:
I think using the toBeInTheDocument() or toBeInTheDom() may be stronger tests in case these values are truthy but still somehow `invalid`

Reply:
Yeah. I spent more time than I had timeboxed for testing, so... next time!